File: Assets/Code/Logic/UI/Tips/View/CommonErrorTipsView.cs
using UnityEngine;
using UnityEngine.UI;

namespace Logic.UI.Tips.View
{
	public class CommonErrorTipsView : MonoBehaviour
	{
		public const string PREFAB_PATH = "ui/tips/common_error_tips_view";

		public Text tipsText;
		private System.Action _callback;
		public static CommonErrorTipsView Open(string tipsString, System.Action callback = null, EUISortingLayer uiSortingLayer = EUISortingLayer.Tips)
		{
			CommonErrorTipsView tips = UIMgr.instance.Open<CommonErrorTipsView>(PREFAB_PATH, uiSortingLayer);
			tips.SetTips(tipsString);
			tips._callback = callback;
			return tips;
		}

		public void SetTips (string tipsString)
		{
			tipsText.text = tipsString;
		}

		public void ClickCloseHandler ()
		{
			if (_callback != null)
			{
				_callback();
			}
			UIMgr.instance.Close(PREFAB_PATH);
		}
	}
}

File: Assets/ToLua/Lua/user/ui/tips/view/common_error_tips_view.lua
local t = {}
local PREFAB_PATH = 'ui/tips/common_error_tips_view'

function t.Open(tipsString)
  local gameObject = Logic.UI.UIMgr.instance:Open(PREFAB_PATH,EUISortingLayer.Tips,UIOpenMode.Overlay)
  t.transform = gameObject:GetComponent(typeof(Transform))
  t.textTips = t.transform:Find('core/img_frame/text_tips'):GetComponent(typeof(Text))
  t.btnOk = t.transform:Find('core/img_frame/btn_ok'):GetComponent(typeof(Button))
  t.btnOk.onClick:AddListener(t.ClickOkBtnHandler)
  
  --set tip
  t.textTips.text = tipsString
end

function t.ClickOkBtnHandler()
  Logic.UI.UIMgr.instance:Close(PREFAB_PATH)
end

return t